plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessor.java

```diff
@@ -538,7 +562,7 @@ public Answer backupSnapshot(final CopyCommand cmd) {
                     final String[] tmp = result.split("#");
                     snapshotBackupUuid = tmp[0];
                     physicalSize = Long.parseLong(tmp[1]);
-                    finalPath = folder + File.separator + snapshotBackupUuid;
+                    finalPath = folder + File.separator + snapshotBackupUuid + ".vhd";
```

Comment:
I am more inclined to reverse this change rather than incur the complexity and testing of an upgrade script.  @syed what is the value of adding the extension?  It is correct, and the way it should have been originally implemented.  However, as @abhinandanprateek points out, this could snapshots created with previous versions of ACS to be inaccessible.